contrib/win32/win32compat/tncon.c

```diff
@@ -148,639 +150,676 @@ ReadConsoleForTermEmul(HANDLE hInput, char *destin, int destinlen)
-							(dwControlKeyState & LEFT_ALT_PRESSED)))
-							NetWriteString2(pParams->Socket, (char *)SHIFT_ALT_PF9_KEY, strlen(SHIFT_ALT_PF9_KEY), 0);
+					}
 
-						else if ((dwControlKeyState & SHIFT_PRESSED) && ((dwControlKeyState & LEFT_CTRL_PRESSED) ||
-							(dwControlKeyState & RIGHT_CTRL_PRESSED)))
-							NetWriteString2(pParams->Socket, (char *)SHIFT_CTRL_PF9_KEY, strlen(SHIFT_CTRL_PF9_KEY), 0);
+					if (isConsoleVTSeqAvailable) {
+						NetWriteString2(pParams->Socket, (char *)octets, n, 0);
 						break;
-					case VK_F10:
-						if (dwControlKeyState == 0)
-							NetWriteString2(pParams->Socket, (char *)PF10_KEY, strlen(PF10_KEY), 0);
-
-						else if (dwControlKeyState == SHIFT_PRESSED)
-							NetWriteString2(pParams->Socket, (char *)SHIFT_PF10_KEY, strlen(SHIFT_PF10_KEY), 0);
+					}
 
-						else if (dwControlKeyState == LEFT_CTRL_PRESSED || dwControlKeyState == RIGHT_CTRL_PRESSED)
-							NetWriteString2(pParams->Socket, (char *)CTRL_PF10_KEY, strlen(CTRL_PF10_KEY), 0);
 
-						else if (dwControlKeyState == LEFT_ALT_PRESSED || dwControlKeyState == RIGHT_ALT_PRESSED)
-							NetWriteString2(pParams->Socket, (char *)ALT_PF10_KEY, strlen(ALT_PF10_KEY), 0);
+					bCapsOn = (inputRecord.Event.KeyEvent.dwControlKeyState & CAPSLOCK_ON);
+					bShift = (inputRecord.Event.KeyEvent.dwControlKeyState & SHIFT_PRESSED);
+					dwControlKeyState = inputRecord.Event.KeyEvent.dwControlKeyState &
+						~(CAPSLOCK_ON | ENHANCED_KEY | NUMLOCK_ON | SCROLLLOCK_ON);
 
-						else if ((dwControlKeyState & SHIFT_PRESSED) && ((dwControlKeyState & RIGHT_ALT_PRESSED) ||
-							(dwControlKeyState & LEFT_ALT_PRESSED)) && ((dwControlKeyState & LEFT_CTRL_PRESSED) ||
-							(dwControlKeyState & RIGHT_CTRL_PRESSED)))
-							NetWriteString2(pParams->Socket, (char *)SHIFT_ALT_CTRL_PF10_KEY, strlen(SHIFT_ALT_CTRL_PF10_KEY), 0);
+					/* ignore the AltGr flags*/
+					if ((dwControlKeyState & dwAltGrFlags) == dwAltGrFlags)
+						dwControlKeyState = dwControlKeyState & ~dwAltGrFlags;
 
-						else if ((dwControlKeyState & RIGHT_ALT_PRESSED) || (dwControlKeyState & LEFT_ALT_PRESSED) &&
-							((dwControlKeyState & LEFT_CTRL_PRESSED) || (dwControlKeyState & RIGHT_CTRL_PRESSED)))
-							NetWriteString2(pParams->Socket, (char *)ALT_CTRL_PF10_KEY, strlen(ALT_CTRL_PF10_KEY), 0);
+					modKey = GetModifierKey(dwControlKeyState);
 
-						else if ((dwControlKeyState & SHIFT_PRESSED) && ((dwControlKeyState & RIGHT_ALT_PRESSED) ||
-							(dwControlKeyState & LEFT_ALT_PRESSED)))
-							NetWriteString2(pParams->Socket, (char *)SHIFT_ALT_PF10_KEY, strlen(SHIFT_ALT_PF10_KEY), 0);
+					if (pParams->fLocalEcho)
+						ConWriteString((char *)octets, n);
 
-						else if ((dwControlKeyState & SHIFT_PRESSED) && ((dwControlKeyState & LEFT_CTRL_PRESSED) ||
-							(dwControlKeyState & RIGHT_CTRL_PRESSED)))
-							NetWriteString2(pParams->Socket, (char *)SHIFT_CTRL_PF10_KEY, strlen(SHIFT_CTRL_PF10_KEY), 0);
+					switch (inputRecord.Event.KeyEvent.uChar.UnicodeChar) {
```

Comment:
nit: It would be very good for readability if you moved this code into a helper function, but this is okay for now.

Reply:
Ignoring this, will do the code refactoring as part of another PR.

Reply:
fixed it.